server/jetstream_cluster.go

```diff
@@ -7762,28 +7760,21 @@ func decodeStreamMsg(buf []byte) (subject, reply string, hdr, msg []byte, lseq u
 	return subject, reply, hdr, msg, lseq, ts, sourced, nil
 }
 
-// Helper to return if compression allowed.
-func (mset *stream) compressAllowed() bool {
-	mset.clMu.Lock()
-	defer mset.clMu.Unlock()
-	return mset.compressOK
-}
-
 // Flags for encodeStreamMsg/decodeStreamMsg.
 const (
 	msgFlagFromSourceOrMirror uint64 = 1 << iota
 )
 
 func encodeStreamMsg(subject, reply string, hdr, msg []byte, lseq uint64, ts int64, sourced bool) []byte {
-	return encodeStreamMsgAllowCompress(subject, reply, hdr, msg, lseq, ts, sourced, false)
+	return encodeStreamMsgAllowCompress(subject, reply, hdr, msg, lseq, ts, sourced)
 }
 
 // Threshold for compression.
 // TODO(dlc) - Eventually make configurable.
 const compressThreshold = 8192 // 8k
 
 // If allowed and contents over the threshold we will compress.
-func encodeStreamMsgAllowCompress(subject, reply string, hdr, msg []byte, lseq uint64, ts int64, sourced bool, compressOK bool) []byte {
+func encodeStreamMsgAllowCompress(subject, reply string, hdr, msg []byte, lseq uint64, ts int64, sourced bool) []byte {
```

Comment:
I would have then removed this function and keep the original `encodeStreamMsg()`, except that `encodeStreamMsg()` could have the comment that says that if the content is over the threshold, it will be compressed.

Reply:
I believe they will not be able to sync until all servers upgraded, should not do any data corruption.